src/ts/framework/Context.ts

```diff
@@ -22,6 +22,10 @@ module framework {
         private _rivets;
         private _sharedObjects : {} = {};
         private _systemWindow : util.SystemWindow = new util.SystemWindow();
+        private _tools : { [key : string] : editorcanvas.tools.BaseTool } = {};
```

Comment:
The tools should be a shared object instead of something attached to the context root.